src/compiler/parser.ts

```diff
@@ -1741,6 +1777,8 @@ module ts {
                     return token === SyntaxKind.GreaterThanToken || token === SyntaxKind.OpenParenToken;
                 case ParsingContext.HeritageClauses:
                     return token === SyntaxKind.OpenBraceToken || token === SyntaxKind.CloseBraceToken;
+                case ParsingContext.JSXAttributes:
+                    return token === SyntaxKind.GreaterThanToken || token === SyntaxKind.SlashToken;

```

Comment:
You use "token === SyntaxKind.OpenBraceToken" to determine if you need to start parsing JSXAttributes.  IsListterminator is called if we're in error recovery and we need to stop parsing whatever construct we're currently in.  If we're in error recovery and we see a }  then presumably we want to stop parsing the current JSXAttributes we're in.